### ISSUE TITLE
Extract + extend is unsupported

### ISSUE THREAD

**Issue by titpetric** [h2non/imaginary#118]
Hello,

I'm playing with the extract API. My intent is to provide "letterbox" like output, which would for example, add a color left and right of a 4:3 ratio image to retrieve a 16:9 ratio image, or add it on the top/bottom to convert a 16:9 to a 4:3. The API already supports `extend` and `background` operations, however:

I get the following error regardless of extend/background settings:

`{"message":"Error while processing the image: extract_area: bad extract area","code":1}`

This is occurring as soon as `areawidth > width`. Of course I didn't come to my typical use case where I could have negative top/left values as well. Let's say, you have a 100x100 image and you want to add a 20x20 black outline. I would do it with extract like this:

~~~
&top=-20
&left=-20
&areawidth=140
&areaheight=140
&width=100
&height=100
&extend=background
&background=0,0,0
~~~

Any chance of supporting this?

**Comment by titpetric** [comment]
As for 16:9/4:3 letterboxing, I figured out that `/resize` endpoint with `nocrop=true` works for this. However the example with negative top/left + larger areawidth/height I believe still applies as a valid use case. Of course for me it applies also for crops, because:

1. we like smart crop (https://github.com/jwagner/smartcrop.js/),
2. in the same way we have multiple top/left/areawidth/areaheight within an image used for crops,
3. and the extract will not work with extended backgrounds from these custom crops

If you'd define a 1:1 crop for lets say, a face, but would like to letterbox it, two API requests should be made (crop to extract exact size, resize to add background for a custom ratio output)

**Comment by h2non** [comment]
This should be fixed in `v1.0.8`.
